libexec/rtld-elf/aarch64/reloc.c

```diff
@@ -754,9 +808,38 @@ reloc_non_plt(Obj_Entry *obj, Obj_Entry *obj_rtld, int flags,
 				return (-1);
 			}
 			break;
+#ifdef __CHERI_PURE_CAPABILITY__
+		case R_MORELLO_TLSDESC:
+#else
 		case R_AARCH64_TLSDESC:
+#endif
 			reloc_tlsdesc(obj, rela, where, flags, lockstate);
 			break;
+#ifdef __CHERI_PURE_CAPABILITY__
+		case R_MORELLO_TLS_TPREL128:
```

Comment:
Add another ifdef in the case to avoid duplicating the comment and the code?

Reply:
Yeah, I could.  I think it ends up only being the `where` assignments that differ.